xarray/tests/test_backends.py

```diff
@@ -3343,6 +3344,61 @@ def test_cache_members(self) -> None:
             observed_keys_2 = sorted(zstore_mut.array_keys())
             assert observed_keys_2 == sorted(array_keys + [new_key])
 
+    @requires_dask
+    @pytest.mark.parametrize("dtype", [int, float])
+    def test_zarr_fill_value_setting(self, dtype):
+        # When zarr_format=2, _FillValue sets fill_value
+        # When zarr_format=3, fill_value is set independently
+        # We test this by writing a dask array with compute=False,
+        # on read we should receive chunks filled with `fill_value`
```

Comment:
If `use_zarr_fill_value_as_mask=True`, then this fill value will be decoded to NaN is believe.

Reply:
I'm turning off all decoding to verify we are setting the `fill_value` right. So I have explicitly set this parameter to False.